test/smartwallet/smartWallet.test.ts

```diff
@@ -1,943 +1,537 @@
+            const {smartWallet, owner, worker} = await loadFixture(prepareFixture);
+
+            await smartWallet.initialize(owner.address, fakeToken.address, worker.address, 10, 400000);
+            expect(fakeToken.transfer).to.be.called;
+        });
+
+        it('Should not call transfer on sponsored deployment', async function(){
+            const {smartWallet, utilSigner, worker} = await loadFixture(prepareFixture);
+
+            await smartWallet.initialize(utilSigner.address, fakeToken.address, worker.address, 0, 0);
+
+            expect(fakeToken.transfer).not.to.be.called;
+        });
+
+        it('Should fail to initialize a contract when it is already initialized', async function(){
+            const {smartWallet, owner, worker} = await loadFixture(prepareFixture);
+
+            await smartWallet.initialize(owner.address, fakeToken.address, worker.address, 10, 400000);  
+
+            await expect(
+                smartWallet.initialize(owner.address, fakeToken.address, worker.address, 10, 400000),
+                'Second initialization not rejected'
+            ).to.be.revertedWith('already initialized');
+        });
+
+        it('Should create the domainSeparator', async function () {
+            const  {smartWallet, owner, worker} = await loadFixture(prepareFixture);
+
+            await smartWallet.initialize(owner.address, fakeToken.address, worker.address, 10, 400000);        
+
+            expect(await smartWallet.domainSeparator()).to.be.properHex(64);
+        });
+
+        it('Should fail when the token transfer method fails', async function () {
+            const  {smartWallet, owner, worker} = await loadFixture(prepareFixture);
+
+            fakeToken.transfer.returns(false);
+
+            await expect(
+                smartWallet.initialize(owner.address, fakeToken.address, worker.address, 10, 400000),
+                'Deployment should be reverted'
+            ).to.be.revertedWith('Unable to pay for deployment');
+        });
+    });
+
+    describe('Function verify()', function(){
+        const HARDHAT_CHAIN_ID = 31337;
+        const FAKE_PRIVATE_KEY = 'da1294b386f1c04cd3276ef3298ef122f226472a55f241d22f924bdc19f92379';
+
+        let fakeToken: FakeContract;
+        let smartWallet: SmartWallet;
+        let privateKey: Buffer;
+        let worker: SignerWithAddress;
+        let externalWallet: Wallet;
+
+        async function prepareFixture(){
+            const smartWalletFactory =await hardhat.getContractFactory('SmartWallet');
+            const smartWallet = await smartWalletFactory.deploy();
+            const [, worker, utilSigner] = await hardhat.getSigners();
+
+            const provider = hardhat.getDefaultProvider();
+            externalWallet =  hardhat.Wallet.createRandom();
+            externalWallet.connect(provider);
+
+            privateKey = Buffer.from(externalWallet.privateKey.substring(2, 66), 'hex');
```

Comment:
Im not sure if this should be inside the fixture or in the beforeEach itself.

Reply:
I moved this to the beforeEach.